lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -90,30 +75,50 @@ sub list_ajax {
         prefetch => [qw(children parents)],
     )->all;
 
-    # need to use all as the order is too complex for a cursor
-    my $comment_count = $self->prefetch_comment_counts([map { $_->id } @jobs]);
+    my $comment_data = $self->schema->resultset('Comments')->comment_data_for_jobs(\@jobs, {bugdetails => 1});
     my @list;
     for my $job (@jobs) {
         my $job_id = $job->id;
+        my %comment_data;
+        if (my $cd = $comment_data->{$job_id}) {
+            $comment_data{comments}     = $cd->{comments};
+            $comment_data{reviewed}     = $cd->{reviewed};
+            $comment_data{label}        = $cd->{label};
+            $comment_data{comment_icon} = $self->comment_icon($job_id, $comment_data{comments});
+            my $bugs = $cd->{bugs};
+            $comment_data{bugs} = [
+                map {
+                    my $bug = $cd->{bugdetails}->{$_};
+                    +{
+                        id        => $bug->id,
+                        content   => $_,
+                        closed    => $bug->open,
+                        title     => $self->bugtitle_for($_, $bug),
+                        url       => $self->bugurl_for($_),
+                        css_class => $self->bugicon_for($_, $bug),
+                    };
+                } sort keys %$bugs
+            ];
+        }
```

Comment:
I consider this block rather complicated and would favor to have functions with good names extracted for something like this so that there is less indention. My preference is rather simple, similar to the code style in the Linux kernel: not more than two levels of indention

Reply:
Now it's in its own function